src/mpn_extras/broadwell/mulhigh_n_basecase.asm

```diff
@@ -0,0 +1,310 @@
+dnl  X64-64 mpn_mullo_basecase optimised for Intel Broadwell.
+
+dnl  Contributed to the GNU project by Torbjorn Granlund.
+
+dnl  Copyright 2017 Free Software Foundation, Inc.
+
+dnl  This file is part of the GNU MP Library.
+dnl
+dnl  The GNU MP Library is free software; you can redistribute it and/or modify
+dnl  it under the terms of either:
+dnl
+dnl    * the GNU Lesser General Public License as published by the Free
+dnl      Software Foundation; either version 3 of the License, or (at your
+dnl      option) any later version.
+dnl
+dnl  or
+dnl
+dnl    * the GNU General Public License as published by the Free Software
+dnl      Foundation; either version 2 of the License, or (at your option) any
+dnl      later version.
+dnl
+dnl  or both in parallel, as here.
+dnl
+dnl  The GNU MP Library is distributed in the hope that it will be useful, but
+dnl  WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+dnl  or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+dnl  for more details.
+dnl
+dnl  You should have received copies of the GNU General Public License and the
+dnl  GNU Lesser General Public License along with the GNU MP Library.  If not,
+dnl  see https://www.gnu.org/licenses/.
+dnl
+dnl   Copyright (C) 2024 Albin Ahlbäck
+dnl
+dnl   This file is part of FLINT.
+dnl
+dnl   FLINT is free software: you can redistribute it and/or modify it under
+dnl   the terms of the GNU Lesser General Public License (LGPL) as published
+dnl   by the Free Software Foundation; either version 3 of the License, or
+dnl   (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+dnl
```

Comment:
@fredrik-johansson Can you make sure that the license here is has been copypasta correctly?